packages/logging/lokiLogProvider_test.go

```diff
@@ -421,6 +422,241 @@ func TestLokiLogProvider_getLogs(t *testing.T) {
 
 }
 
+func TestLokiLogProvider_queryLogs(t *testing.T) {
+	t.Run("should query logs", func(t *testing.T) {
+		// given
+		svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			assert.Equal(t, "/loki/api/v1/query_range?direction=backward&end=1655722130600667903&limit=1000&query=%7Bpod%3D~%22test.%2A%22%7D&start=1653130130600667903", r.URL.String())
+			username, password, ok := r.BasicAuth()
+			assert.True(t, ok)
+			assert.Equal(t, "admin", username)
+			assert.Equal(t, "admin123", password)
+			_, err := w.Write(lokiResponseTestData)
+			require.NoError(t, err)
+		}))
+		defer svr.Close()
```

Comment:
Instead of creating a httptest server for each test case, you could use the http client as member of llp instead of using the DefaultClient. The default client also has the problem, that it does not set a timeout what can produce never ending requests...

Reply:
I added an httpClient as member of the lokiLogProvider with a default-timeout of 30 seconds.
But I leave the  httptest servers in the test-cases as is, because every test-server has ist own handler with its own assertions.